Content.Shared/Clothing/EntitySystems/ClothingSystem.cs

```diff
@@ -92,12 +92,14 @@ private void ToggleVisualLayers(EntityUid equipee, HashSet<HumanoidVisualLayers>
             InventorySystem.InventorySlotEnumerator enumerator = _invSystem.GetSlotEnumerator(equipee);
 
             bool shouldLayerShow = true;
-            while (enumerator.NextItem(out EntityUid item))
+            while (enumerator.NextItem(out EntityUid item, out SlotDefinition? slot))
             {
                 if (TryComp(item, out HideLayerClothingComponent? comp))
                 {
                     if (comp.Slots.Contains(layer))
                     {
+                        if (slot.SlotFlags == SlotFlags.POCKET) break;
```

Comment:
this should either be
- checking the Slots field on `ClothingComponent`
- checking a new field on `HideLayerClothingComponent`

just hardcoding it to the pockets isn't the correct way to do it. Ideally the second option is done so that clothing objects that fit into multiple slots can specify when they want to hide hair.

Reply:
Additionally, the `GetSlotEnumerator` has an actual method that takes in flags to filter the enumerator, so you can just use that instead of manually checking the flags on each slot.

Reply:
> Additionally, the `GetSlotEnumerator()` method takes in a flags argument to filter the enumerator, so you can just use that instead of manually checking the flags on each slot.
That's a good point, but as far as I've tested the flags argument doesn't affect the InventorySlotEnumerator it returns. And there are suspicions that the `GetSlotEnumerator()`method doesn't need the flags argument at all, at least in the current implementation.

Reply:
@ElectroJr When sending flags argument in `GetSlotEnumerator()` we just replace the SlotFlags argument in InventorySlotEnumerator, I honestly don't even understand what for.